### ISSUE TITLE
Database Issues when file path includes a space

### ISSUE THREAD

**Issue by dashwav** [Frederikam/FredBoat#464]
**FredBoat Stats (Selfhosted)**
```
JDA version:                    3.5.0_332
FredBoat version:               2.3_DEV
Lavaplayer version:             1.2.47
```
When running the software, if the file path has a space in it, FredBoat can't load the db package.

```
[14:35:37] [ ERROR] [] [DatabaseConnection] Failed to load entity package fredboat.db.entity.main
space.npstr.sqlsauce.DatabaseException: fredboat.db.entity.main (null) does not appear to be a valid package
        at space.npstr.sqlsauce.DatabaseConnection.getClassesForPackage(DatabaseConnection.java:545)
        at space.npstr.sqlsauce.DatabaseConnection.access$000(DatabaseConnection.java:77)
        at space.npstr.sqlsauce.DatabaseConnection$1.lambda$getManagedClassNames$0(DatabaseConnection.java:413)
        at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1380)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at space.npstr.sqlsauce.DatabaseConnection$1.getManagedClassNames(DatabaseConnection.java:419)
        at org.hibernate.jpa.boot.internal.PersistenceUnitInfoDescriptor.getManagedClassNames(PersistenceUnitInfoDescriptor.java:98)
        at org.hibernate.jpa.boot.internal.StandardJpaScanEnvironmentImpl.<init>(StandardJpaScanEnvironmentImpl.java:31)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.populate(EntityManagerFactoryBuilderImpl.java:789)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.<init>(EntityManagerFactoryBuilderImpl.java:215)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.<init>(EntityManagerFactoryBuilderImpl.java:163)
        at org.hibernate.jpa.boot.spi.Bootstrap.getEntityManagerFactoryBuilder(Bootstrap.java:32)
        at org.hibernate.jpa.boot.spi.Bootstrap.getEntityManagerFactoryBuilder(Bootstrap.java:89)
        at org.hibernate.jpa.HibernatePersistenceProvider.getEntityManagerFactoryBuilder(HibernatePersistenceProvider.java:176)
        at org.hibernate.jpa.HibernatePersistenceProvider.createContainerEntityManagerFactory(HibernatePersistenceProvider.java:151)
        at space.npstr.sqlsauce.DatabaseConnection.<init>(DatabaseConnection.java:198)
        at space.npstr.sqlsauce.DatabaseConnection$Builder.build(DatabaseConnection.java:845)
        at fredboat.db.DatabaseManager.main(DatabaseManager.java:93)
        at fredboat.main.Launcher.main(Launcher.java:110)
Caused by: java.io.FileNotFoundException: /home/deploy/fredboat%202/FredBoat.jar (No such file or directory)
        at java.util.zip.ZipFile.open(Native Method)
        at java.util.zip.ZipFile.<init>(ZipFile.java:225)
        at java.util.zip.ZipFile.<init>(ZipFile.java:155)
        at java.util.jar.JarFile.<init>(JarFile.java:166)
        at java.util.jar.JarFile.<init>(JarFile.java:103)
        at space.npstr.sqlsauce.DatabaseConnection.getClassesForPackage(DatabaseConnection.java:521)
```
the filepath here was `/home/deploy/fredboat 2/FredBoat.jar` as soon as I switched it to `/home/deploy/fredboat2/FredBoat.jar` everything went back to normal. It looks like it is percent encoding the filepath instead of escaping the special characters.

`Caused by: java.io.FileNotFoundException: /home/deploy/fredboat%202/FredBoat.jar (No such file or directory)`

**Comment by napstr** [comment]
Whats the issue here? You got a well readable error telling you what you did wrong, and resolved the issue easily.

**Comment by dashwav** [comment]
I mean the issue is that the bot can't handle whitespace in the filepath? Do you not want bug reports if  people can find a workaround on their own?

Not handling whitespace isn't a user issue, it's a software issue.

**Comment by napstr** [comment]
Selfhosters are notoriously known for abusing the hell out of formatting and whitespaces in all places. I've grown extremely indifferent to those issues. But you are right, this is clearly bug. Please accept my apology. I've opened a report with upstream.

**Comment by napstr** [comment]
Fixed in 23a92cdf3e5eb24ac83b7b8f92b6f89947b9b376
